### DISCUSSION TITLE
How to connect node to edge label

### DISCUSSION THREAD

**Discussion by zayonoah** [xyflow/xyflow#2421] [Q&A] [answered]
Hello,

I am trying to connect a node to an edge label. Is this something that is possible? 

For example from my screenshot, I am wanting to attach my 'note' node to the edge label with 'Fiber Span 31'. 

I am able to add a handle to my edge labels, but it appears that creating edges only works when both the target and source are nodes. Any ideas on how to make this work with the target being an edge and not a node? 

If that is not possible, my plan was to remove the custom edge labels and position nodes in the same place. The drawback with that is then the 'edge label' (which is now a node made to look like an edge label) isn't actually connected to the edge. So if the edge line moves, the node/edge label won't move with it, which isn't ideal. Help is much appreciated, thank you!

**Comment by moklick** [comment, marked as answer]
This is not possible. You can only connect nodes with nodes. However you could create a node as an edge label. I created an example for this: https://codesandbox.io/s/kind-platform-rgdtyk?file=/src/EdgeNode.tsx 

Connect the two nodes, and you can see that a custom node will be rendered as an edge label.

**Reply by zayonoah** [reply]
Thank you for the quick answer! That is a great example and should work perfect for my needs. Thanks again
